pkg/scheduler/listers.go

```diff
@@ -52,3 +54,35 @@ func (f FakePodLister) ListPods(s labels.Selector) (selected []api.Pod, err erro
 	}
 	return selected, nil
 }
+
+// ServiceLister interface represents anything that can list services for a scheduler.
+type ServiceLister interface {
+	// TODO: make this exactly the same as client's ListServices() method...
+	ListServices() (api.ServiceList, error)
+	// gets the service that has the selector that can match the labels on the given pod
+	GetPodService(api.Pod) (api.Service, error)
```

Comment:
Based on a suggestion by @mikedanese, I have modified the interface to return multiple services. It is upto the predicate to handle the result. Currently, it just picks the first service in that list.